designs/consolidation-policies.md

```diff
@@ -0,0 +1,72 @@
+
+Karpenter's consolidation is an all or nothing offering today. Nodepools can disable consolidation and let only empty nodes be removed, or customers can opt into all forms of consolidation.
+
+## Design
+
+### Option 1: Add price thresholds, merge consolidation controllers
+The motivation for the additional controls is the high cost of disruption of certain workloads for small or negative cost savings. So instead of offering controls, the proposal is to add price improvement thresholds for consolidation, to avoid this scenario.
+
+The proposal is to find a price improvement threshold which accurate represents the "cost" of disrupting the workloads. This threshold could be an arbitrary-fixed value (e.g. 20% price improvement), or heuristically computed by Karpenter based on the cluster shape (e.g. nodepool ttl's might be a heuristic input).
+
+The separation of the single-node and multi-node consolidation is arbitrary. The single-node consolidation finds the "least costly" eligible node to disrupt, and uses that as a solution, even if it's a poor solution. However, the least-costly eligible node to disrupt is not necessarily the most cost-savings node to disrupt, but the single-node consolidation stops after it has any solution.
+
+In contrast, the multi-node consolidation spends as much time as possible to find a solution. So while price thresholds could be implemented into each consolidation controller, it is simpler to consider merging their responsabilities into one shared controller based on multi-node consolidation. This combined consolidation controller could search as long as possible for the most effective consolidation outcome, from emptiness, multi-node or single-node replacements.
+
+#### Pros/Cons
+* 👍👍 No new configuration for the `NodePool` crd
+* ~ More complex implementation for consolidation, but easier to understand (only 1 controller for consolidation).
+* 👎 It's possible a one-size fits all price threshold will satisfy some customers and not others
+* 👎 High amount of engineering effort to implement.
+
+### Option 2: Refactor and Expand Consolidation Policies
+Another approach would be to not make significant changes to consolidation, but expose controls to allow enabling or disabling various consolidation controllers. Karpenter already presents the binary consolidation options. The `NodePool` resource has a disruption field that exposes the `consolidationPolicy` field. This field is an enum with two supported values: `WhenEmpty` and `WhenUnderutilized`.
+
+The proposal is to add two new consolidation policy enum values:
+
+* `WhenCheaper`
+* `WhenUnderutilizedOrCheaper`
+
+And change the semantics of the existing consolidation policy enum value:
+
+* `WhenUnderutilized`
```

Comment:
Changing the semantic of the existing consolidation policy to resolve to something different is a breaking change, and could definitely surprise people who upgrade without changing their values to the right one. If we're going this route, I'd prefer we keep the existing enum val to mean the same thing, and add another one to portray the mode you're proposing